### PR TITLE
Diverge3 - Workman Layout

### PR DESCRIPTION
**Pull request by leothelocust**
The [workman programming layout](https://github.com/ojbucao/workman) for the Diverge3 keyboard.

### CODE REVIEW

File: keyboards/diverge3/config.h [hunk 1]
```diff
@@ -51,50 +53,14 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
 #define DEBOUNCING_DELAY 5
 
-/* define if matrix has ghost (lacks anti-ghosting diodes) */
-//#define MATRIX_HAS_GHOST
-
-/* number of backlight levels */
-
-/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
-#define LOCKING_SUPPORT_ENABLE
```
Comment (drashna) [inline keyboards/diverge3/config.h]: Any reason for removing the code here?

File: keyboards/diverge3/keymaps/workman/config.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/diverge3/keymaps/workman/config.h]: Could you replace the boiler place with just `#pragma once`? 

Eg, the file should look like: 
```c
#pragma once

// place overrides here
#define PERMISSIVE_HOLD
#define TAPPING_TERM 150
```

And the include is not necessary, as they keyboard's config. includes it already

File: keyboards/diverge3/keymaps/workman/keymap.c [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/diverge3/keymaps/workman/keymap.c]: These includes aren't necessary. You should only need the `QMK_KEYBOARD_H` include, usually.

File: keyboards/diverge3/config.h (hunk 1 above)
Comment (leothelocust) [reply keyboards/diverge3/config.h]: My mistake, I didn't think it applied, so I'm adding it back.

File: keyboards/diverge3/keymaps/workman/config.h (hunk 2 above)
Comment (leothelocust) [reply keyboards/diverge3/keymaps/workman/config.h]: Done.

File: keyboards/diverge3/keymaps/workman/keymap.c (hunk 3 above)
Comment (leothelocust) [reply keyboards/diverge3/keymaps/workman/keymap.c]: Done.

**Comment by leothelocust** [comment]
@drashna Please hold off on merging this.  After making the changes you suggested, I cannot get my keyboard to function correctly.  I'll report back later today

**Comment by drashna** [comment]
Not a problem.

I'm guessing that some of these settings may help with that: https://docs.qmk.fm/#/config_options?id=behaviors-that-can-be-configured

**Comment by leothelocust** [comment]
@drashna So this _was_ working before I started making the changes you suggested.  Temporarily I reverted all my code to commit `d869079` and now when I flash, my two keyboard halfs are mirrored, so "a" is "i" and "left shift" is "right shift" etc...

My mind is hurting right now. Any thoughts?

**Comment by leothelocust** [comment]
I did try `#define MASTER_LEFT` and then `MASTER_RIGHT` in `workman/config.h` to no avail.

**Comment by drashna** [comment]
Do the other keymaps work correctly?

**Comment by leothelocust** [comment]
No, my keyboard seems to be permanently _flipped_ now...

**Comment by drashna** [comment]
Well, you could add an alternate matrix define in the diverge's h file.   

That way, you can have both the normal and flipped version.  
https://github.com/qmk/qmk_firmware/blob/60b6749d3b8a4e26fdc7c3ae4a9e37152cc15080/keyboards/diverge3/diverge3.h#L31


Also, it may be worth converting the board to the Split Common code.  That ... may help too.

**Comment by drashna** [comment]
I'm going to go ahead and merge this now.  

If you're still having issues with the board, let me know.

**Pull request merged**